paymentsheet/src/test/java/com/stripe/android/paymentsheet/model/SupportedPaymentMethodTest.kt

```diff
@@ -199,6 +200,70 @@ class SupportedPaymentMethodTest {
         assertThat(getPMsToAdd(mockIntent, null, lpmRepository)).isEqualTo(expected)
     }
 
+    @Test
+    fun `Test supported payment method filters us bank account when FC SDK not available`() {
+        val lpmRepository = LpmRepository(
+            arguments = LpmRepository.LpmRepositoryArguments(
+                resources = ApplicationProvider.getApplicationContext<Application>().resources,
+            ),
+            lpmInitialFormData = LpmRepository.LpmInitialFormData(),
+        ).apply {
+            initializeWithPaymentMethods(
+                mapOf(
+                    PaymentMethod.Type.Card.code to card,
+                    PaymentMethod.Type.USBankAccount.code to LpmRepository.hardCodedUsBankAccount
+                )
+            )
+        }
+        val mockIntent = mock<PaymentIntent>()
+        whenever(mockIntent.paymentMethodTypes).thenReturn(listOf("card", "us_bank_account"))
+        whenever(mockIntent.isLiveMode).thenReturn(false)
+        whenever(mockIntent.unactivatedPaymentMethods).thenReturn(listOf("card"))
+
+        val expected = listOf<SupportedPaymentMethod>().plus(card)
+
+        assertThat(
+            getPMsToAdd(
+                stripeIntent = mockIntent,
+                config = PaymentSheet.Configuration("Test", allowsDelayedPaymentMethods = true),
+                lpmRepository = lpmRepository,
+                isFinancialConnectionsAvailable = { false },
+            )
+        ).isEqualTo(expected)
+    }
+
+    @Test
+    fun `Test supported payment method contains us bank account when FC SDK available`() {
+        val lpmRepository = LpmRepository(
+            arguments = LpmRepository.LpmRepositoryArguments(
+                resources = ApplicationProvider.getApplicationContext<Application>().resources,
+            ),
+            lpmInitialFormData = LpmRepository.LpmInitialFormData(),
+        ).apply {
+            initializeWithPaymentMethods(
+                mapOf(
+                    PaymentMethod.Type.Card.code to LpmRepository.HardcodedCard,
+                    PaymentMethod.Type.USBankAccount.code to LpmRepository.hardCodedUsBankAccount
+                )
+            )
+        }
+        val mockIntent = mock<PaymentIntent>()
+        whenever(mockIntent.paymentMethodTypes).thenReturn(listOf("card", "us_bank_account"))
+        whenever(mockIntent.isLiveMode).thenReturn(false)
+        whenever(mockIntent.unactivatedPaymentMethods).thenReturn(listOf("card"))
```

Comment:
We should avoid mocks for objects that we control. Can you use `PaymentIntentFactory` instead?